### ISSUE TITLE
Error running coverage report, traced back to `tao` dependency `core-video-sys`

### ISSUE THREAD

**Issue by mihirsamdarshi** [tauri-apps/tao#270]
Hey guys, so I was having some issues compiling sometimes, especially when building my app in order to run `grcov`, my app was having some trouble on macOS, getting the following error message:

```
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-m64" "-L" "/Users/user/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" ... [long linker command here] ...
  = note: Undefined symbols for architecture x86_64:
            "_CFMutableAttributedStringGetTypeID", referenced from:
                _$LT$core_foundation..attributed_string..CFMutableAttributedString$u20$as$u20$core_foundation..base..TCFType$GT$::type_id::h9f6f71bdd347aca0 in foo-cef9f22f855ec251.198s3ihm7i58x886.rcgu.o
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: aborting due to previous error
```

I did some digging and found [this issue](https://github.com/servo/core-foundation-rs/pull/357) - which mentioned that the error was related to an older version of `core-foundation-rs`. 

I did a little more digging and found that `tao` depends on `core-video-sys`, which in turn depends on that older version of `core-foundation-rs`. [That `core-video-sys` crate](https://github.com/LuoZijun/rust-core-video-sys) seems to be abandoned. [Someone graciously cleaned up the code and dependencies](https://github.com/kvark/rust-core-video-sys/tree/cleanup) and submitted a pull request but this has not been merged for nearly 5 months.

While investigating, I was hoping that the PR would fix the issue, and it did in fact. In order to test it I cloned `tao`, and added [a patch statement](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html) to the `Cargo.toml` to use the PR branch. I then added patch statement to `wry` to reference the local `tao` copy with the patch statement in its `Cargo.toml`, and so on and so forth for `tauri-runtime-wry` and `tauri`, and finally my application.

I was wondering how/if the team could address this - perhaps submodule the crate or something? From my understanding crates aren't allowed to be published with git dependencies, but are there any other solutions?

Thanks for all your work, and as always, please let me know how I can help!

**Comment by nothingismagick** [comment]
What a great finding. Thankyou so much for digging in!

Now what to do about it is the next question. Sometimes we adopt packages (either through forking and publishing under a similar name but in the hopes the original author picks it back up) or we accept the transfer and become maintainers. The latter is more work, because we have to apply our own CI and licenses etc.

In either case, it's worthwhile tracking down the original authors and asking them how they would like to proceed. If they don't respond in a couple weeks then we can assume its been abandoned, and we are unlikely to receive a proper repo transfer. 

In the meantime, we can prepare the fork(s). Would you consider doing this on your own and then transferring to our github org? And reat assured, we would happily welcome you as a maintainer. :)

**Comment by mihirsamdarshi** [comment]
@nothingismagick I would love to help out in any way, this project is so useful to me. I will create a fork of that fixed version of `core-video-sys`. please let me know what you need for me to do before a potential adoption of the crate

**Comment by wusyong** [comment]
`next` branch doesn't seem to use `core-video-sys` anymore. @mihirsamdarshi Could you try if latest commit work for you?

**Comment by mihirsamdarshi** [comment]
> `next` branch doesn't seem to use `core-video-sys` anymore. @mihirsamdarshi Could you try if latest commit work for you?

I am not sure what you mean, I still do see it [listed in `Cargo.toml`](https://github.com/tauri-apps/tao/blob/f8866a084afe18983c91c67401d8759697578a9c/Cargo.toml#L64-L67)?

**Comment by wusyong** [comment]
Ah sorry, silly me just run cargo tree on different platform.
I think we can make a soft fork from kvark's one until the PR is merged.

**Comment by wusyong** [comment]
This should be fixed by #274. Let me know if it's still an issue.

**Comment by mihirsamdarshi** [comment]
thanks @wusyong. I have a couple of notes. On looking at the source on [`doc.rs`](https://docs.rs/crate/tao-core-video-sys/0.2.0/source/) and crates.io I still see a dependency on the `obj_c` crate, which is really only used by one file (`open_gl_es_texture_cache.rs`), I was wondering if you saw that as necessary? [I forked the `core-video-sys` crate](https://github.com/mihirsamdarshi/rust-core-video-sys) that I mentioned and added a few changes, I was wondering if you wanted to take a look at that?
